### PR TITLE
Log Unauthorized requests

### PR DESCRIPTION
**Pull request by corylanou**
Fixes https://github.com/influxdb/influxdb/issues/1444

### CODE REVIEW

File: server_test.go [hunk 1]
```diff
@@ -96,6 +97,47 @@ func TestServer_DeleteDataNode(t *testing.T) {
...
 
+// Test unuathorized requests logging
+func TestServer_UnauthorizedRequests(t *testing.T) {
+	s := OpenServer(NewMessagingClient())
+	defer s.Close()
+
+	s.SetAuthenticationEnabled(true)
+
+	var b bytes.Buffer
+	s.SetLogOutput(&b)
```
Comment (otoolep) [inline server_test.go]: Cute. :-)

**Comment by otoolep** [comment]
+1 on green build.

**Comment by dgnorton** [comment]
Why move Authorize func into the server?  `ExecuteQuery` could log the error returned from `Authorize`.

My personal preference is for libraries never to log but everyone may not agree on that one.

**Comment by otoolep** [comment]
@dgnorton -- I tend to agree, but it seems to me that we're not dealing with libraries here, so much as "packages". It's just a word, granted, but a "package" is a bigger deal if you follow me. Also, if we don't have any logging in our packages, we'll have no logging at all.

Let me know if I am missing your point.

**Comment by dgnorton** [comment]
@otoolep  It's more than just a word.  In a Go, anything in `package main` is an app and everything else compiles into a library.  I'm saying only `package main` should log.  I have a long list of reasons why libraries shouldn't log.  :)

- Libraries don't know where or how they're going to be used.  In my app, maybe I only want to use a specific logging framework or logging format.  That gets messed up if a library I'm using has its own logging.

- Even if I own all the apps and libraries, let's say I decide the format needs to change or I want to swap out the logging frame work, now I have to go through thousands of lines of logging code.

- Thousands of lines of logging code - ugh.  Who wants to look at that or maintain it.  It's cognitive load that doesn't need to be there.

- Moving functions from one package to another when refactoring becomes a pain in the ass because the logging works slightly differently in that package, or the config options are different, or that package doesn't even have logging.

- Apps that don't want to log still pay the cost

- Logging can change the behavior of the code  (e.g., somebody decides the logging needs to use synchronization when writing to stdout, etc.).

Logging at app level keeps the logging in one spot.  The rest of the code doesn't bloat and slow down with logging over time.  The logging format is uniform.  You don't have to worry about adding logging calls every time you check an error or bubble it up...logging just magically works because you handled errors.  It's rainbows and unicorns all the way up.

**Comment by corylanou** [comment]
I think the comments around logging make sense.  The are outside of the scope of this PR though.  We should open a new issue that discusses logging as a larger concern and decide how we want to proceed from there.  For now, I'm merging this and can revisit once the larger issues have been settled.

**Pull request merged**